BusRoutes/View/RouteDetails.xaml.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Xamarin.Forms;
+using BusRoutes.Model;
+using BusRoutes.ViewModel;
+
+namespace BusRoutes.View
+{
+	public partial class RouteDetails : ContentPage
+	{
+		public RouteDetails (Route route)
+		{
+			InitializeComponent ();
+			this.BindingContext = new RouteDetailsViewModel (route);
+		}
+
+		private void StreetListView_OnItemSelected(object sender, SelectedItemChangedEventArgs e)
+		{
+			StreetListView.SelectedItem = null;
+		}
+
+		private void TimetableListView_OnItemSelected(object sender, SelectedItemChangedEventArgs e)
+		{			
+			Navigation.PushAsync(new RouteDepartures(((RouteDetailsViewModel)this.BindingContext).CurrentRoute, (DepartureSummary)TimetableListView.SelectedItem ));
```

Comment:
For clarity you could extract `((RouteDetailsViewModel)this.BindingContext)` into a property `private RouteDetailsViewModel ViewModel { get { return ((RouteDetailsViewModel)this.BindingContext); } }`